### PR TITLE
Add tool for exporting firestore uuid tables

### PR DESCRIPTION
**Pull request by as2388**
Uses new https://github.com/AfricasVoices/Pipeline-Infrastructure/pull/41

### CODE REVIEW

File: id_infrastructure/export_firestore_uuid_tables.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    export = dict()  # of table_name -> {mappings: dict of data -> uuid}
+    for i, table_name in enumerate(table_names):
+        log.info(f"Fetching mappings from table {i + 1}/{len(table_names)}: {table_name}...")
+        mappings = id_tables.get_table(table_name, None).get_all_mappings()
+        export[table_name] = {
+            "mappings": mappings
+        }
+        log.info(f"Fetched {len(mappings)} mappings")
+
+    log.info(f"Writing mappings to '{export_file_path}'...")
```
Comment (IsaackMwenda) [inline id_infrastructure/export_firestore_uuid_tables.py:51]: Shouldn't we write this directly to the backup bucket?  Am a bit nervous about writing this locally but could be persuaded that we need to cross-check before uploading.

**Review by IsaackMwenda — COMMENTED** [review]
Am a bit nervous about exporting this locally but could be persuaded that we need to cross-check before uploading?

**Comment by as2388** [comment]
> Am a bit nervous about exporting this locally but could be persuaded that we need to cross-check before uploading?

Good point, it was only done like this to mirror the strategy used for Coda backups. I'll rework this slightly to accept --export-file-path and --gcs-upload-path args like we used for archiving Rapid Pro, so that we can back-up using memory/temporary directories only unless the user explicitly opts-in to having a local export created.

**Comment by as2388** [comment]
@IsaackMwenda I've reworked to make local exports optional, and to do all the download, compression and upload in memory otherwise.

File: id_infrastructure/export_firestore_uuid_tables.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    if gzip_export_file_path is not None:
+        log.warning(f"Writing mappings to local disk at '{gzip_export_file_path}'...")
+        with open(gzip_export_file_path, "wb") as f:
+            f.write(export_compressed)
+
+    if gcs_upload_path is not None:
+        log.info(f"Uploading the mappings to {gcs_upload_path}...")
+        google_cloud_utils.upload_string_to_blob(google_cloud_credentials_file_path, gcs_upload_path,
+                                                 export_compressed)
+    log.info("Done")
```
Comment (IsaackMwenda) [inline id_infrastructure/export_firestore_uuid_tables.py:73]:
```suggestion
    log.info(f"{table_names} uuid export complete.")
```

**Review by IsaackMwenda — APPROVED** [review]

File: id_infrastructure/export_firestore_uuid_tables.py (hunk 2 above)
Comment (as2388) [reply id_infrastructure/export_firestore_uuid_tables.py:73]: Changed to log.info(f"Export complete ({len(table_names)} table(s))") (we already printed each table name above)

**Pull request merged**